Desafio3/Desafio/Desafio.Models/Consulta.cs

```diff
@@ -6,26 +6,108 @@
+        ///<list type="bullet">
+        ///<item>o formato HHMM;</item>
+        ///<item>ser de um período futuro;</item>
+        ///<item>estar no limite do horário de funcionamento do consultório, 8:00h às 19:00h; e</item>
+        ///<item>ser definida sempre de 15 em 15 minutos(Ex. 1400, 1730, 1615).</item>
+        ///</list>
+        ///</param>
+        ///<param name = "dataHoraFinal">Representa o valor da propriedade <see cref="DataHoraFinal"/>, 
+        ///<br>A <see langword="data"/> deve possuir:</br>
+        ///<list type="bullet">
+        ///<item>formato DD/MM/AAAA;</item>
+        ///<item>e ser de um período futuro.</item>
+        ///</list>
+        ///<br>A <see langword="hora final"/> deve possuir:</br>
+        ///<list type="bullet">
+        ///<item>o formato HHMM;</item>
+        ///<item>ser de um período futuro;</item>
+        ///<item>ser de um período futuro maior que a propriedade <paramref name="dataHoraInicial"/>;</item>
+        ///<item>estar no limite do horário de funcionamento do consultório, 8:00h às 19:00h; e</item>
+        ///<item>ser definida sempre de 15 em 15 minutos(Ex. 1400, 1730, 1615).</item>
+        ///</list>
+        ///</param>
+        public Consulta(Paciente paciente, DateTime dataHoraInicial, DateTime dataHoraFinal)
+        {
+            this.Paciente = paciente;
+            this.DataHoraInicial = dataHoraInicial;
+            this.DataHoraFinal = dataHoraFinal;
+        }
+
+        /// <summary>
+        /// Sobreescreve método <see langword="ToString"/>.
+        /// </summary>
+        /// <returns>Uma <see langword="string"/> contendo dados do <see cref="Consulta"/>.</returns>
+        public override string ToString()
+        {
+            return $"{this.DataHoraInicial:t} "
+                 + $"{this.DataHoraFinal:t} "
+                 + $"{this.Tempo:hh\\:mm} " 
+                 + $"{this.Paciente.Nome} "
+                 + $"{this.Paciente.DtNascimento:d}\n";
+        }
+
+        /// <summary>
+        /// Sobreescreve método <see langword="Equals"/>.        
+        /// </summary>
+        /// <param name="obj">Representa um objeto de <see cref="Consulta"/>.</param>
+        /// <returns>
+        /// <list type="bullet">
+        /// <item>
+        /// Retorna <see langword="true"/> 
+        /// <list type="bullet">
+        /// <item>se a <see cref="Consulta"/> tiver o mesmo <see cref="Paciente"/>.</item>
+        /// <item>se a <see cref="Consulta"/> tiver a mesma <see cref="DataHoraInicial"/>.</item>       
+        /// </list>
+        /// </item>
+        /// <item>
+        /// Retorna <see langword="false"/> caso contrário.
+        /// </item>
+        /// </list>
+        /// </returns>
+        public override bool Equals(object? obj)
+        {
+            return obj is Consulta consulta &&
+                   EqualityComparer<Paciente>.Default.Equals(this.Paciente, consulta.Paciente) &&
```

Comment:
Essa linha é equivalente a `this.Paciente.Equals(consulta.Paciente)` ?

Se o efeito for o mesmo tá tranquilo!

Reply:
Sim é o mesmo.